bpfd-operator/apis/v1alpha1/kprobeProgram_types.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope=Cluster
+
+// KprobeProgram is the Schema for the KprobePrograms API
+// +kubebuilder:printcolumn:name="SectionName",type=string,JSONPath=`.spec.sectionname`
+// +kubebuilder:printcolumn:name="NodeSelector",type=string,JSONPath=`.spec.nodeselector`
+// +kubebuilder:printcolumn:name="FunctionName",type=string,JSONPath=`.spec.func_name`,priority=1
+// +kubebuilder:printcolumn:name="Offset",type=integer,JSONPath=`.spec.offset`,priority=1
+// +kubebuilder:printcolumn:name="RetProbe",type=boolean,JSONPath=`.spec.retprobe`,priority=1
+// +kubebuilder:printcolumn:name="Namespace",type=string,JSONPath=`.spec.namespace`,priority=1
+type KprobeProgram struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec KprobeProgramSpec `json:"spec"`
+	// +optional
+	Status KprobeProgramStatus `json:"status,omitempty"`
+}
+
+// KprobeProgramSpec defines the desired state of KprobeProgram
+// +kubebuilder:printcolumn:name="FunctionName",type=string,JSONPath=`.spec.func_name`
+// +kubebuilder:printcolumn:name="Offset",type=integer,JSONPath=`.spec.offset`
+// +kubebuilder:printcolumn:name="RetProbe",type=boolean,JSONPath=`.spec.retprobe`
+// +kubebuilder:printcolumn:name="Namespace",type=string,JSONPath=`.spec.namespace`
+type KprobeProgramSpec struct {
+	BpfProgramCommon `json:",inline"`
+
+	// Functions to attach the kprobe to.
+	FunctionNames []string `json:"func_names"`
+
+	// Offset added to the address of the function for kprobe.
+	// Not allowed for kretprobes.
+	// ANF-TODO: figure out how to validate this requirement.
+	// +optional
+	// +kubebuilder:default:=0
+	Offset uint64 `json:"offset"`
+
+	// Whether the program is a kretprobe.  Default is false
+	// +optional
+	// +kubebuilder:default:=false
+	RetProbe bool `json:"retprobe"`
+
+	// Namespace to attach the uprobe in.
```

Comment:
Correct.  Namespace is currently ignored by bpfd.  But I was wondering whether we wanted to do anything about it in the k8s code.  If we don't, as soon as we support it in bpfd, it would just start working for k8s without changes.  I do have a check in bpfctl.  I'm fine either way.

Reply:
Can you just comment this field out here until it's supported?